### ISSUE TITLE
[Bug]:  Auto commit and push stops working nearly everyday

### ISSUE THREAD

**Issue by Samuelodan** [denolehov/obsidian-git#462]
### Describe the bug

I often notice that there has been no update from the plugin for a while and I see that its last Git update was hours ago. So I have to disable and enable the plugin for to resume normal operation. This wasn't the case when I initially installed the plugin but it's been happening for a few months now.

I have included the error I found from the logs.
The screenshot below shows the error go away after toggling the disable plugin button and the app proceeds to commit and push changes I made over 24 hours ago. I hope this is descriptive enough.
Thanks.
<img width="554" alt="Screenshot 2023-03-13 at 03 40 38" src="https://user-images.githubusercontent.com/35481602/224596536-ba963536-6b52-4b44-9803-9bebe0130e8d.png">

### Relevant errors (if available)

```shell
git obsidian error: Connection reset by 140.82.121.4 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

VM148 plugin:obsidian-git:23985 Uncaught (in promise) Error: Connection reset by 140.82.121.4 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

    at Object.action (plugin:obsidian-git:23985:25)
    at PluginStore.exec (plugin:obsidian-git:24010:25)
    at eval (plugin:obsidian-git:21571:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (plugin:obsidian-git:21569:16)
    at GitExecutorChain.eval (plugin:obsidian-git:21553:44)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-git:20402:24)
```

### Steps to reproduce

It just happens.

### Expected Behavior

I expect it to work consistently to back up changes.

### Operating system

macOS

### Installation Method

None

### Plugin version

2.17.4

**Comment by Wit4r7** [comment]
Have the same issue on multiple macs. Started happening with the latest version. It is constantly committing 0 files.  The issue is only with the auto-commit/push. Manual still works fine.

**Comment by Vinzent03** [comment]
@Wit4r7 So your issue is not authentication when pushing, but committing of zero files? Then this is probably a different issue. Is just the notification wrong or do you really have empty commits, which I really doubt, because git doesn't allow that by default.

**Comment by Wit4r7** [comment]
@Vinzent03 Yeah it is committing zero files. I downgraded and updated again. Currently, I'm unable to reproduce the error, but I'm also not on my main pc. If the error occurs again, I will create a new ticket and provide additional information.

**Comment by Samuelodan** [comment]
> Have the same issue on multiple macs. Started happening with the latest version. It is constantly committing 0 files.  The issue is only with the auto-commit/push. Manual still works fine.  

Interesting. I noticed it also happens after the laptop has been put to sleep for a while.
The file changes counter remains at zero and it says the last Git update was a while ago (it just happened some mins ago and last Git update was 19 hours ago). It's like it stopped checking for changes and didn't resume even after the laptop was woken.

**Comment by NicklasWallgren** [comment]
I've also encountered this issue, it stops syncing after the laptop has returned from sleep.
